packages/mask/src/plugins/IdeaMarket/apis/index.ts

```diff
@@ -0,0 +1,162 @@
+                id
+                name
+                tokenID
+                market {
+                    id
+                    marketID
+                    name
+                }
+                rank
+                latestPricePoint {
+                    price
+                }
+                supply
+                holders
+                daiInToken
+                dayChange
+            }
+        }`,
+        variables: {
+            searchTerm,
+            rowsPerPage,
+            skip: page === 0 ? 0 : page * rowsPerPage,
+            filters,
+        },
+    }
+    const response = await fetch(SUBGRAPH_URL, {
+        body: JSON.stringify(body),
+        method: 'POST',
+    })
+
+    const res: IdeaToken[] = (await response.json())?.data.ideaTokens
+
+    return res
+}
+
+export async function fetchUserTokensBalances(holder: string) {
+    const body = {
+        query: `query IdeaTokenBalances ($holder: String!) {
+            ideaTokenBalances(where: {holder: $holder}){
+                token {
+                    id
+                    name
+                    market {
+                        id
+                        name
+                        marketID
+                    }
+                    latestPricePoint {
+                      price
+                    }
+                    daiInToken
+                    dayChange
+                }
+                id
+                holder
+                amount
+            }
+        }`,
+        variables: { holder },
+    }
+    const response = await fetch(SUBGRAPH_URL, {
+        body: JSON.stringify(body),
+        method: 'POST',
+    })
+
+    const res = (await response.json())?.data.ideaTokenBalances
+
+    return res
+}
+
+export async function fetchTwitterLookup(token: IdeaToken) {
+    const response = await fetch(
+        `https://api.twitter.com/2/users/by/username/${token.name.slice(1)}?user.fields=profile_image_url`,
+        {
+            headers: {
+                Authorization: `Bearer ${TWITTER_BEARER_TOKEN}`,
+            },
+        },
+    )
+    const res = (await response.json())?.data
+
+    return res
+}
+
+export async function fetchTwitterLookups(tokens: IdeaToken[]) {
+    const twitterHandles = tokens
+        .filter((token) => token.market.marketID === Markets.Twitter)
+        .map((token) => token.name.slice(1))
+        .join(',')
+    const response = await fetch(
+        `https://api.twitter.com/2/users/by?usernames=${twitterHandles}&user.fields=profile_image_url`,
+        {
+            headers: {
+                Authorization: `Bearer ${TWITTER_BEARER_TOKEN}`,
+            },
+        },
+    )
+    const twitterLookups = (await response.json())?.data
+
+    // create a hashmap in order to optimize twitter username lookups
+    // eslint-disable-next-line unicorn/no-array-reduce
```

Comment:
please rewrite this part to avoid reducing.

Reply:
Done.